### ISSUE TITLE
Loading BN weights for models with different sequence max-length

### ISSUE THREAD

**Issue by PiranjaF** [fchollet/keras#4134]
I have a pretrained model for sentiment analysis that was trained on texts with maximum length `m_1` and now I want to put the model in production for texts with maximum length `m_2`. How do I easily do that?

I've tried changing `m_1` to `m_2` in yaml model definition file before loading weights, but that seems to be a too hacky approach as my model now outputs only NaNs.

**Comment by PiranjaF** [comment]
Could this be due to the use of BatchNormalization layers?

**Comment by PiranjaF** [comment]
```
I've found that it is due to the BN layers. A reproducible example can be found here:

from __future__ import print_function
import numpy as np
np.random.seed(1337)  # for reproducibility

from keras.preprocessing import sequence
from keras.utils import np_utils
from keras.models import Model, Sequential
from keras.layers import Dense, Dropout, Activation, Embedding
from keras.layers import Input, LSTM, SimpleRNN, GRU
from keras.layers.embeddings import Embedding
from keras.layers.recurrent import LSTM
from keras.layers.convolutional import Convolution1D
from keras.layers.normalization import BatchNormalization
from keras.layers.pooling import GlobalMaxPooling1D, GlobalAveragePooling1D
from keras.regularizers import l2
from keras.datasets import imdb

max_features = 20000
batch_size = 32

def pad_define_train(maxlen, save=None, load=None):
    (X_train, y_train), (X_test, y_test) = imdb.load_data(nb_words=max_features)
    X_train = sequence.pad_sequences(X_train, maxlen=maxlen)
    X_test = sequence.pad_sequences(X_test, maxlen=maxlen)

    model_input = Input(shape=(maxlen,), dtype='int32')
    x = Embedding(max_features, 16, input_length=maxlen)(model_input)
    x = BatchNormalization(mode=0,
                           axis=1,
                           gamma_regularizer=l2(1E-4),
                           beta_regularizer=l2(1E-4))(x)
    x = Activation('relu')(x)
    x = Convolution1D(16, 3,
                      init="he_uniform",
                      border_mode="same",
                      bias=False,
                      W_regularizer=l2(1E-4))(x)
    x = GlobalAveragePooling1D()(x)
    x = Dense(1)(x)
    x = Activation('sigmoid')(x)
    model = Model(input=[model_input], output=[x])

    model.compile(loss='binary_crossentropy',
                  optimizer='adam',
                  metrics=['accuracy'])

    if load is not None:
        model.load_weights(load)
        model.evaluate(X_test, y_test,
                       batch_size=batch_size)

    if save is not None:
        model.fit(X_train, y_train, batch_size=batch_size, nb_epoch=1,
                  validation_data=(X_test, y_test))
        model.save_weights(save)

pad_define_train(80, save='temp_weights.hdf')
pad_define_train(200, load='temp_weights.hdf')
```
